### PR TITLE
[thor] new port #1319

### PR DESCRIPTION
**Pull request by atkawa7**

### CODE REVIEW

File: ports/thor/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/src/*.lib"
+    "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/*/src/*.lib"
+)
+if(DLLS)
+    file(INSTALL ${DLLS} DESTINATION ${CURRENT_PACKAGES_DIR}/bin)
+endif()
+file(INSTALL ${LIBS} DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
+if(DEBUG_DLLS)
+    file(INSTALL ${DEBUG_DLLS} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
+endif()
```
Comment (codicodi) [inline ports/thor/portfile.cmake]: Thor provides install target. Why not use it? Also, you should call `vcpkg_copy_pdbs()` here.

**Comment by codicodi** [comment]
Thor headers seem to depend on Aurora headers, yet the latter are not installed. When building an example application I'm getting:

`fatal error C1083: Cannot open include file: 'Aurora/Config.hpp': No such file or directory`

File: ports/thor/portfile.cmake (hunk 1 above)
Comment (atkawa7) [inline ports/thor/portfile.cmake]: @codicodi I tried it. It seems that when I do that. The build always fails. Could there a reason for that?

Comment (codicodi) [inline ports/thor/portfile.cmake]: I just tried and it looks like errors are caused by includes in debug directory (just remove them to fix this) and licenses in root directory (you could move them to /share/thor and rename one of them to `copyright` I guess).

**Comment by codicodi** [comment]
I've noticed Aurora headers are installed along with Thor ones when install target is used. AFAIK no other popular library uses Aurora so maybe there is no need to add independent port?

**Comment by atkawa7** [comment]
@codicodi Thanks for your suggestion. Just wondering how you would deal with dll. For example if I use your suggestions I get  `DLLs without any exports are likely a bug in the build script.`

```file(READ ${CONFIG_FILE} CONFIG_H)
   if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
       string(REPLACE "defined(SFML_STATIC)" "1" CONFIG_H "${CONFIG_H}")
   else()
       string(REPLACE "defined(SFML_STATIC)" "0" CONFIG_H "${CONFIG_H}")
   endif()

file(WRITE ${CONFIG_FILE} "${CONFIG_H}")
```

**Comment by atkawa7** [comment]
@codicodi Thoughts?

**Comment by codicodi** [comment]
It looks like internal copy of Aurora is still used during build. I added `#error` to one of it's headers and build instantly failed. Is there a way to use vcpkg provided Aurora instead?

**Comment by atkawa7** [comment]
@codicodi Added

**Comment by codicodi** [comment]
Works fine now, thanks!

**Comment by atkawa7** [comment]
@ras0219-msft Could you review?

**Comment by ras0219-msft** [comment]
Thanks both of you, looks good to me!

**Pull request merged**